### ISSUE TITLE
WINGS: FileMetadata and versioning

### ISSUE THREAD

**Issue by elrayle** [samvera/hyrax#3923]
### Descriptive summary

The move to support valkyrie as the backend for Hyrax requires some reworking of the versioning for files.

### Know places that need to be addressed

Other places may also need to be changed. But the following are known changes that are required.

#### Hyrax::FileMetadata resource (under app/models)

* add an attributes to track past versions (e.g. `attribute :version_ids, ::Valkyrie::Types::Set`)
* #versions method which returns all versions... something like

```
    def versions
      query_service = Valkyrie.config.metadata_adapter.query_service
      query_service.find_references_by(resource: self, property: :version_ids)
      # As written, this will return the ids.  So depending on use cases for this method, may need to create the resource objects for each id
    end
```

#### Hyrax::VersioningService

* #perform_create_through_valkyrie method.  It is pending completion of #indexing_adapter method.

#### Wings::FileMetadataBuilder service (under lib/wings/services)

* #attach_file_metadata_to_valkyrie_file_set method.  It makes a copy of passed in file_metadata excluding the id and at this writing member_ids.  
  * By not copying the id, it is making a new version.  Need to verify that this works as expected once other versioning code is updated.
  * The member_ids attribute was originally planned to hold the ids of the various versions.  So, if it is correct to exclude past version ids here, whatever attribute was added to Hyrax::FileMetadata to hold the version ids (e.g. version_ids), is what should be excluded.  It is possible that member_ids will have already been removed from that code by the time this issue is addressed.  Also, there is an open question about how the version ids will get set in the new version being created by FileMetadataBuilder.

**Comment by no-reply** [comment]
@rjkati is file versioning in the Hyrax QA script? i'm wondering whether there's prior work here, or why it hasn't been flagged as a release blocker so far.

**Comment by rjkati** [comment]
@no-reply file versioning is included in the QA script. To pass, users need to be able to upload a new version and restore a prior version. 

I suspect that file versioning had not yet been tested, since filesets were not working properly until very recently.

**Comment by rjkati** [comment]
Per discussion at DSU, we are closing this ticket and opening new tickets for specific file versioning issues. 

Uploading a new version is addressed in #6211 . Restoring a prior version cannot be tested until #6211 is completed.
